### PR TITLE
[TS] LPS-166886

### PR DESCRIPTION
**Pull request by crodriguezy**

### CODE REVIEW

**Comment by catarineaguiar** [comment]
@crodriguezy no regressions found.

**Comment by dantewang** [comment]
@crodriguezy Could you please add a test case to cover this?

**Comment by crodriguezy** [comment]
Hi @dantewang , I just pushed a new commit with the integration test

**Comment by crodriguezy** [comment]
ci:test:sf

**Comment by crodriguezy** [comment]
ci:test:relevant

**Comment by crodriguezy** [comment]
ci:test:relevant

File: portal-web/docroot/html/taglib/ui/input_localized/page.jsp [hunk 1]
```diff
@@ -116,9 +116,9 @@
 				languageValue = ParamUtil.getString(request, name + StringPool.UNDERLINE + curLanguageId, languageValue);
 			}
 
-			if (curLanguageId.equals(defaultLanguageId) && Validator.isNull(languageValue)) {
+			/*if (curLanguageId.equals(defaultLanguageId) && Validator.isNull(languageValue)) {
 				languageValue = LocalizationUtil.getLocalization(xml, defaultLanguageId, true);
-			}
+			}*/
 		%>
```
Comment (yuhai) [inline portal-web/docroot/html/taglib/ui/input_localized/page.jsp:123]: @crodriguezy, this looks weird. Since the change is in the tag, except with the current ticket, does other usages exist the issue? It looks unrelated with the current issue. If the issue still exists in other usages, please create one new ticket to fix it.

File: modules/apps/expando/expando-test/src/testIntegration/java/com/liferay/expando/service/test/ExpandoValueLocalServiceTest.java [hunk 2]
```diff
@@ -169,6 +170,32 @@ public void testAddNondefaultLocalizedStringValue() throws Exception {
...
+		ExpandoColumn column = ExpandoTestUtil.addColumn(
+			_expandoTable, "Test Column",
+			ExpandoColumnConstants.STRING_ARRAY_LOCALIZED);
+
+		ExpandoValue value = ExpandoTestUtil.addValue(
+			_expandoTable, column,
+			HashMapBuilder.put(
+				_frLocale, "TestFR"
+			).build());
+
```
Comment (yuhai) [inline modules/apps/expando/expando-test/src/testIntegration/java/com/liferay/expando/service/test/ExpandoValueLocalServiceTest.java:188]: @crodriguezy, ExpandoTestUtil.addValue() will throw one exception. So the test can't be passed. Please see the error stack
```
com.liferay.expando.service.test.ExpandoValueLocalServiceTest > testAddSiteDefaultLocalizedStringValue FAILED
    java.lang.ClassCastException: java.lang.String cannot be cast to [Ljava.lang.String;
        at com.liferay.portlet.expando.model.impl.ExpandoValueImpl.setStringArrayMap(ExpandoValueImpl.java:525)
        at com.liferay.portlet.expando.service.impl.ExpandoValueLocalServiceImpl.addValue(ExpandoValueLocalServiceImpl.java:404)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at com.liferay.portal.spring.aop.AopMethodInvocationImpl.proceed(AopMethodInvocationImpl.java:50)
        at com.liferay.portal.spring.transaction.TransactionInterceptor.invoke(TransactionInterceptor.java:69)
        at com.liferay.portal.spring.aop.AopMethodInvocationImpl.proceed(AopMethodInvocationImpl.java:57)
        at com.liferay.portal.spring.aop.AopInvocationHandler.invoke(AopInvocationHandler.java:49)
        at com.sun.proxy.$Proxy173.addValue(Unknown Source)
        at com.liferay.portlet.expando.service.impl.ExpandoValueLocalServiceImpl.addValue(ExpandoValueLocalServiceImpl.java:584)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at com.liferay.portal.spring.aop.AopMethodInvocationImpl.proceed(AopMethodInvocationImpl.java:50)
        at com.liferay.portal.spring.transaction.TransactionInterceptor.invoke(TransactionInterceptor.java:69)
        at com.liferay.portal.spring.aop.AopMethodInvocationImpl.proceed(AopMethodInvocationImpl.java:57)
        at com.liferay.portal.spring.aop.AopInvocationHandler.invoke(AopInvocationHandler.java:49)
        at com.sun.proxy.$Proxy173.addValue(Unknown Source)
        at com.liferay.expando.kernel.service.ExpandoValueLocalServiceUtil.addValue(ExpandoValueLocalServiceUtil.java:225)
        at com.liferay.portlet.expando.util.test.ExpandoTestUtil.addValue(ExpandoTestUtil.java:65)
        at com.liferay.portlet.expando.util.test.ExpandoTestUtil.addValue(ExpandoTestUtil.java:87)
        at com.liferay.expando.service.test.ExpandoValueLocalServiceTest.testAddSiteDefaultLocalizedStringValue(ExpandoValueLocalServiceTest.java:183)
```

File: portal-impl/src/com/liferay/portlet/expando/service/impl/ExpandoColumnLocalServiceImpl.java [hunk 3]
```diff
@@ -540,7 +540,7 @@ else if (type == ExpandoColumnConstants.STRING_LOCALIZED) {
 			Map<Locale, String> defaultValuesMap =
 				(Map<Locale, String>)defaultData;
 
-			Locale defaultLocale = LocaleUtil.getDefault();
+			Locale defaultLocale = LocaleUtil.getSiteDefault();
```
Comment (yuhai) [inline portal-impl/src/com/liferay/portlet/expando/service/impl/ExpandoColumnLocalServiceImpl.java:544]: @crodriguezy, I think the change is not right. Since ExpandoColumn doesn't belong to site level and it belongs to company level. We can't add ExpandoColumn in any sites.

**Comment by yuhai** [comment]
@crodriguezy, please see my above comments and resend it.

Thanks,
Hai

File: portal-web/docroot/html/taglib/ui/input_localized/page.jsp (hunk 1 above)
Comment (crodriguezy) [reply portal-web/docroot/html/taglib/ui/input_localized/page.jsp:123]: I've separeted it in a new bug, please see https://github.com/liferay-core-infra/liferay-portal/pull/2398
